typing/typecore.ml

```diff
@@ -4640,22 +4676,22 @@ and type_application env funct sargs =
             (l,ty,lv) :: !eliminated_optional_arguments;
           Some (fun () -> option_none env (instance ty) Location.none)
         in
-        let remaining_sargs, arg =
+        let remaining_sargs, arg, arg_loc =
```

Comment:
Personal nitpicking 2: the `arg` argument requires some computation whereas `arg_loc` is straightforward to compute. It may make sense to commute those two elements of the tuple so you can decrease the diff size.

Reply:
Paired with the `arg` now. The location needs to be passed because `arg` is wrapped in a thunk doing side effects.